mods/sgssb/abilities.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+exports.BattleAbilities = {
+	//Ashley the Pikachu
+	primalsurge: {
+		name: 'Primal Surge',
+		id: 'primalsurge',
+		onStart: function (source) {
+			this.setWeather('primordialsea');
+			this.setTerrain('electricterrain');
+			this.terrainData.duration = 0;
+		},
+		onAnySetWeather: function (target, source, weather) {
+			if (this.getWeather().id === 'primordialsea' && !(weather.id in {desolateland:1, primordialsea:1, deltastream:1})) return false;
+		},
+		onAnySetTerrain: function (target, source, terrain) {
```

Comment:
Yeah this event dosen't exist, I tried to add one to fix a bug with seeds but Zarel had a better way to do it.